core/src/main/java/io/questdb/griffin/engine/table/AllTablePartitionsCursorFactory.java

```diff
@@ -0,0 +1,55 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2023 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.table;
+
+import io.questdb.cairo.CairoEngine;
+import io.questdb.cairo.TableToken;
+import io.questdb.cairo.sql.RecordMetadata;
+import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.std.ObjHashSet;
+import io.questdb.std.ObjList;
+
+public class AllTablePartitionsCursorFactory extends SequentialRecordCursorFactory<ShowPartitionsRecordCursorFactory> {
+
+    @Override
+    public RecordMetadata getMetadata() {
+        return ShowPartitionsRecordCursorFactory.METADATA;
+    }
+
+    @Override
```

Comment:
Each ShowPartitionsRecordCursorFactory stores a number of objects (most notably TableReaders) that are used briefly .  If the number of tables is big, factory and cursor could allocate quite a lot and keep reallocating on each call .
I reckon it'd be better to:
- fetch and  keep the list of table tokens on getCursor()
- reuse a single instance ShowPartitionsRecordCursorFactory internally
- in hasNext() method iterate over list of table tokens and re-initialize ShowPartitionsRecordCursorFactory when current instance is exhausted